vmware/cluster-api-provider-cloud-director
# [VCDA-3636] Nvidia GPU support for containerd in cloud init script

- Addition of Nvidia runtime installation in cloud init script
- Installation and configuration of Nvidia plugin in containerd conf file
- Common cloud init template to reduce duplication between node types
- Conditional logic in cloud init template to handle presence of gpu support
- Support of common template inside vcdmachine controller

Testing done:
- compare cloud init output from common template for both control plane node and worker node with those generated without these changes
- create clusters with nvidiaGPU flag set and not set to ensure correct cloud init template is used
- ensure that containerd changes take effect and that containerd is able to start successfully
- following commands were run on vm to ensure containerd changes were picked up successfully

```
ctr image pull docker.io/nvidia/cuda:11.0-base
ctr run --rm --gpus 0 -t docker.io/nvidia/cuda:11.0-base cuda-11.0-base nvidia-smi
```

for template reference: https://pkg.go.dev/text/template


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/152)
